src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -317,11 +323,18 @@ protected boolean compileAdditionalSources()
             FileOps.copyResource(getAdditionalSourceFileLocation()  + fileName, workDir.resolve(fileName));
             processBuilder.command().add(fileName);
         }
+        
+        Path nativeCodeDir = getNativeCodePath();
+        for( String fileName: getNativeCodeList() ) {
+            FileOps.copyFile(nativeCodeDir.resolve(fileName), workDir.resolve(fileName));
+            processBuilder.command().add(fileName);
```

Comment:
this would also bring the header files to the compile command. We need to copy the header files, but they should not be in the compile command line.
e.g. we need `gcc -I/path/to/headers hello.c` 
with `hello.c` having `#include hello.h` and hello.h needs to be at path/to/headers

I just found out that the current code we have for this is wrong (it uses the header file in the gcc argument, which is not what we want)

Reply:
Ah I see. I was relying on current code for reference. 
This should be fixed in latest commit.